### ISSUE TITLE
Attributes on macro does not trigger error or warning

### ISSUE THREAD

**Issue by topecongiro** [rust-lang/rust#47771]
Using `rustc 1.25.0-nightly (9fd7da904 2018-01-25)`,

```rust
fn foo() {
    println!("hello world");
}

fn main() {
    #[rustfmt_skip]
    foo ();
}
```

the abvoe code triggers an error:

```
error[E0658]: The attribute `rustfmt_skip` is currently unknown to the compiler and may have meaning added to it in the future (see issue #29642)
  --> test.rs:18:5
   |
18 |     #[rustfmt_skip]
   |     ^^^^^^^^^^^^^^^
   |
   = help: add #![feature(custom_attribute)] to the crate attributes to enable

error: aborting due to previous error
```

However, the following code compiles without any error:

```rust
fn main() {
    #[rustfmt_skip]
    println! ("hello world");
}
```

It looks like the attributes on macros are entirely ignored from the compiler. Is this an expected behavior?

**Comment by zackmdavis** [comment]
Two potentially relevant comments?—

https://github.com/rust-lang/rust/blob/8ccab7eed5f4fc93500fbf242e575073ca70d7cb/src/libsyntax/visit.rs#L110-L117

https://github.com/rust-lang/rust/blob/8ccab7eed5f4fc93500fbf242e575073ca70d7cb/src/libsyntax/feature_gate.rs#L1101-L1111

**Comment by BGR360** [comment]
This no longer produces a compiler error ([playground](https://play.rust-lang.org/?version=nightly&mode=debug&edition=2015&gist=1cc802d2dc5f2dd9635d2a022441436c)):

```rust
fn foo() {
    println!("hello world");
}

fn main() {
    #[rustfmt::skip]
    foo ();
    
    #[rustfmt::skip]
    println! ("hello world");
}
```

```
hello world
hello world
```

Can we close?
